### ISSUE TITLE
Build Fails with librsvg after Openssl Update

### ISSUE THREAD

**Issue by danyeaw** [wingtk/gvsbuild#468]
As reported by @zs332,  5aa8593 updated to Openssl 1.1.1m which caused librsvg to fail to build:

```
NMAKE : fatal error U1077: 'C:\gtk-build\tools\cargo\bin\cargo.EXE' : return code '0x65'
Stop.
Traceback (most recent call last):
File "c:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 493, in build
if self.build_one(p):
File "c:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 624, in build_one
skip_deps = proj.build()
File "c:\gtk-build\github\gvsbuild\gvsbuild\projects.py", line 1248, in build
self.exec_vs(cmd)
File "c:\gtk-build\github\gvsbuild\gvsbuild\utils\base_project.py", line 110, in exec_vs
self.builder.exec_vs(cmd, working_dir=self._get_working_dir(), add_path=add_path)
File "c:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 871, in exec_vs
self.execute(self.sub_vars(cmd), working_dir=working_dir, add_path=add_path, env=self.vs_env)
File "c:\gtk-build\github\gvsbuild\gvsbuild\utils\builder.py", line 948, in __execute
subprocess.check_call(args, cwd=working_dir, env=env, shell=True)
File "C:\Python39\lib\subprocess.py", line 373, in check_call
raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'nmake -f makefile.vc CFG=release PREFIX=C:\gtk-build\gtk\x64\release PYTHON=C:\gtk-build\tools\python.3.10.0\tools\python.exe install INTROSPECTION=1' returned non-zero exit status 2.
Error: librsvg build failed
```

**Comment by mhertz** [comment]
Thanks, but I cannot see openssl even installed as a dep to librsvg, or am I completely blind/dense :)

I hoped to have found the culprit now through your posting, as spend a few days unsuccesfull now - I need revert fontconfig update, as fails with cannot find git, which I have installed from msys and in path, and then librsvg fails(also with openssl reverted as per your post here) - it's linker error with missing bcrypt something symbols in an rsvg dll or lib - I read from other projects that it means cannot find bcrypt.dll from winSDK, so should add to msvc additional deps of project, which i'm not really capable doing through this, so i'm still reverting commits back and fourth and currently trying various librsvg updates/downgrades. I was thinking the pkgconfig 1.8 update was related, as reportedly errors with meson-build as uses : as path seperator or alike, but didn't change anything reverting that. Possibly all my issues is because reverting fontconfig, so probably should look into that first i.e. check all build-files generated.

Anyway, sorry for blabbing on here and off-topic.

**Comment by nacho** [comment]
I think we will need something like this for librsvg: https://github.com/aws/dcv-color-primitives/commit/44cccec36ebeac19b56de0273e9bf17bb8f53f94

**Comment by mhertz** [comment]
Can I just say how much I appreciate you nacho :) All of you of-course. I'm about to add/test and will reply back later. I was just gonna do more tests but luckily rechecked here first, which helps immensely and would not find myself I'm sure. Yesterday tried newest librsvg upstream, previous, and next previous of gvsbuild librsvg, plus --old-librsvg, which worked, but has issues building adwaita-icon-theme, as I see it doesn't correctly copy over the gdk-pixbuff svg loader(so fails translating from svg to png).

Thank you again!

**Comment by mhertz** [comment]
Tested working now from clean build, thank you so much again nacho,really appreciate it! :)
````
diff -urN a/gvsbuild/projects.py b/gvsbuild/projects.py
--- a/gvsbuild/projects.py	2021-12-20 20:42:08.031084200 +0100
+++ b/gvsbuild/projects.py	2021-12-20 20:45:54.456083000 +0100
@@ -1225,6 +1225,7 @@
             ],
             patches = [
                 '0001-keep-cairo-dep.patch',
+                'bcrypt.patch',
             ],
             )
         if Project.opts.enable_gi:
diff -urN a/patches/librsvg/bcrypt.patch b/patches/librsvg/bcrypt.patch
--- a/patches/librsvg/bcrypt.patch	1970-01-01 01:00:00.000000000 +0100
+++ b/patches/librsvg/bcrypt.patch	2021-12-20 19:36:53.783937000 +0100
@@ -0,0 +1,11 @@
+diff -ur a/win32/config-msvc.mak b/win32/config-msvc.mak
+--- a/win32/config-msvc.mak	2021-09-14 02:16:20.000000000 +0200
++++ b/win32/config-msvc.mak	2021-12-20 18:00:56.810592600 +0100
+@@ -118,6 +118,7 @@
+ 	libxml2.lib			\
+ 	advapi32.lib			\
+ 	userenv.lib			\
++	bcrypt.lib			\
+ 	ws2_32.lib
+ 
+ RSVG_PIXBUF_LOADER_CFLAGS =			\
````

**Comment by danyeaw** [comment]
@mhertz Would you be able to please submit a pull request?

**Comment by mhertz** [comment]
Sorry, done, and I apologise for somewhat overtaking your thread, forgot say that before :)

**Comment by danyeaw** [comment]
Closed by #469
